server/src/client_api.rs

```diff
@@ -22,12 +25,15 @@ use unleash_types::client_metrics::{
 #[get("/client/features")]
 pub async fn features(
     edge_token: EdgeToken,
-    features_source: web::Data<dyn EdgeSource>,
+    features_cache: web::Data<DashMap<String, ClientFeatures>>,
 ) -> EdgeJsonResult<ClientFeatures> {
-    features_source
-        .get_client_features(&edge_token)
-        .await
+    features_cache
+        .get(&cache_key(edge_token))
+        .map(|features| features.clone())
         .map(Json)
+        .ok_or_else(|| {
+            EdgeError::PersistenceError("TODO: Don't use this error here!!!!!!!!".into())
```

Comment:
Partially dealt with. I've patched the error message but the error responses need some more thought so that we don't trigger horrible behaviour in the SDKs